client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -461,6 +463,27 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow, stepName } ) => {
 	const { setPendingAction } = useDispatch( ONBOARD_STORE );
 
 	function pickDesign( _selectedDesign: Design | undefined = selectedDesign ) {
+		const shouldGoToAssembler =
+			_selectedDesign?.is_virtual &&
+			_selectedDesign?.slug === BLANK_CANVAS_DESIGN.slug &&
+			isDesktop &&
+			isEnabled( 'pattern-assembler/dotcompatterns' );
+
+		if ( shouldGoToAssembler ) {
+			const assemblerDesign = {
+				..._selectedDesign,
+				design_type: 'assembler',
+			} as Design;
+
+			setSelectedDesign( assemblerDesign );
```

Comment:
I don't think we need it since we don't do that during the normal site setup flow:

https://github.com/Automattic/wp-calypso/blob/9ce211ef8474966426bbe9a95980932fbc4cfed3/client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx#L526-L548